### PR TITLE
Facia presses 'lite' versions of fronts as well as full versions

### PR DESCRIPTION
**Pull request by QuarpT**
## What does this change?

Facia presses 'lite' versions of fronts as well as full versions. For most requests, Facia does not need an full pressed front. This is an optimisation.

I will change the base branch to master after the dependent feature branch is merged.

## What is the value of this and can you measure success?

Facia can process smaller Json files, more performant - it will be able to handle a higher request rate.

## Tested in CODE?

I will

### CODE REVIEW

File: common/app/layout/slices/Container.scala [hunk 1]
```diff
@@ -31,6 +32,16 @@ object Container extends Logging {
   def fromConfig(collectionConfig: CollectionConfig): Container =
     resolve(collectionConfig.collectionType)
 
+  def maxStories(collectionConfig: CollectionConfig, items: Seq[PressedContent]): Option[Int] = {
```
Comment (TBonnin) [inline common/app/layout/slices/Container.scala]: Would `storiesCount` would be a better name for this function?

File: facia-press/app/frontpress/FapiFrontPress.scala [hunk 2]
```diff
@@ -112,10 +115,9 @@ trait FapiFrontPress extends Logging {
     val stopWatch: StopWatch = new StopWatch
 
     val pressFuture = getPressedFrontForPath(path)
-      .map { pressedFront: PressedPage =>
-        val json: String = Json.stringify(Json.toJson(pressedFront))
-        FaciaPressMetrics.FrontPressContentSize.recordSample(json.getBytes.length, new DateTime())
-        putPressedJson(path, json)
+      .map { pressedFronts: PressedPageVersions =>
+        putPressedPage(path, pressedFronts.full, FullType)
+        putPressedPage(path, pressedFronts.lite, LiteType)
```
Comment (TBonnin) [inline facia-press/app/frontpress/FapiFrontPress.scala]: what happen if one of the put fails?

File: facia-press/app/controllers/Application.scala [hunk 3]
```diff
@@ -23,6 +23,7 @@ class Application(liveFapiFrontPress: LiveFapiFrontPress, draftFapiFrontPress: D
 
   def generateLivePressedFor(path: String): Action[AnyContent] = Action.async { request =>
     liveFapiFrontPress.getPressedFrontForPath(path)
+      .map(_.full)
```
Comment (TBonnin) [inline facia-press/app/controllers/Application.scala]: Should we add a route like `/pressed/live/*path.lite` so we can get the lite version of the pressed front?

**Review by TBonnin — APPROVED** [review]
I made some small comments but it looks good to me. I think this is going to be a huge improvement. Thank you

File: facia-press/app/frontpress/FapiFrontPress.scala (hunk 2 above)
Comment (QuarpT) [reply facia-press/app/frontpress/FapiFrontPress.scala]: Then the two pressed files will not match and the error will be logged.

This shouldn't be a problem. It's already possible for the pressed page to be updated between the browser retrieving the initial pressed page, and the show-more request.

File: facia-press/app/controllers/Application.scala (hunk 3 above)
Comment (QuarpT) [reply facia-press/app/controllers/Application.scala]: Can do, I will add it

**Comment by TBonnin** [comment]
See https://github.com/guardian/frontend/pull/18365

**Pull request closed without merging**